aws/resource_aws_vpc_endpoint_service.go

```diff
@@ -214,17 +242,39 @@ func resourceAwsVpcEndpointServiceRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("error setting allowed_principals: %s", err)
 	}
 
+	err = d.Set("private_dns_name_config", flattenPrivateDnsNameConfiguration(svcCfg.PrivateDnsNameConfiguration))
+	if err != nil {
+		return fmt.Errorf("error setting private_dns_name_config: %w", err)
+	}
+
 	return nil
 }
 
+func flattenPrivateDnsNameConfiguration(privateDnsNameConfiguration *ec2.PrivateDnsNameConfiguration) []interface{} {
+	if privateDnsNameConfiguration == nil {
+		return []interface{}{}
```

Comment:
To prevent unexpected behaviors with Terraform CLI in the future, this should `return nil` here to prevent a mismatched configuration and saved state.

```suggestion
		return nil
```